contracts/ETHOrders.sol

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.18;
+
+import { EIP712 } from "@openzeppelin/contracts/utils/cryptography/EIP712.sol";
+import "@1inch/solidity-utils/contracts/libraries/SafeERC20.sol";
+import "@1inch/solidity-utils/contracts/OnlyWethReceiver.sol";
+
+import "./interfaces/IPostInteraction.sol";
+import "./OrderLib.sol";
+
+/// @title ETH limit orders contract
+contract ETHOrders is IPostInteraction, OnlyWethReceiver {
+    using SafeERC20 for IWETH;
+    using OrderLib for IOrderMixin.Order;
+    using ConstraintsLib for Constraints;
+    using ExtensionLib for bytes;
+    using AddressLib for Address;
+
+    error AccessDenied();
+    error ETHTransferFailed();
+    error InvalidOrder();
+    error NotEnoughBalance();
+    error ExistingOrder();
+
+    address private immutable _limitOrderProtocol;
+    IWETH private immutable _WETH; // solhint-disable-line var-name-mixedcase
+    // Tracks ETH balances for orders.
+    mapping(bytes32 => uint256) public ethOrderBalance;
```

Comment:
We could merge these two mapping into one until someone try to swap 79 billions of ETH, which is limitation of `uint96` :)